From 144f87ef4c9f6f84dac42da4c4dc37d03fa35fc9 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 1 Sep 2016 09:49:18 +0300
Subject: [PATCH 0613/1345] icu: dts: cp110: move interrupts of devices to be
 ICU

commit  8fdc1cba7e45d53f1544808d9b96c1c3c3ccb5e4 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch update the interrupts of the devices to use
ICU dynamic interrupt allocation.

Change-Id: I3e6898e05cde07fe153bbc06ea94443c1924374c
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32424
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi    |   65 +++++++-----
 .../boot/dts/marvell/armada-cp110-master.dtsi      |   14 +--
 .../arm64/boot/dts/marvell/armada-cp110-slave.dtsi |    6 +-
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi      |  103 +++++++++++---------
 4 files changed, 107 insertions(+), 81 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 830b22a..524e761 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -16,6 +16,8 @@
 /*
  * Generic Device Tree describing Marvell Armada CP-110 device
  */
+#include <dt-bindings/interrupt-controller/mvebu-icu.h>
+
 rtc@284000 {
 	compatible = "marvell,armada8k-rtc";
 	reg = <0x284000 0x20>, <0x284080 0x24>;
@@ -43,6 +45,17 @@ cps_syscon0: system-controller@440000 {
 		"cps-usb3dev", "cps-eip150", "cps-eip197";
 };
 
+icu_s: interrupt-controller@1E0000 {
+	compatible = "marvell,icu";
+	reg = <0x1E0000 0x10>;
+	#interrupt-cells = <3>;
+	#size-cells = <1>;
+	#address-cells = <0>;
+	icu-spi-base = <0>;
+	interrupt-controller;
+	interrupt-parent = <&gic>;
+	gicp-spi = <0x0 0xF03F0040 0x0 0xF03F0048>;
+};
 pinctrl@440000 {
 	reg = <0x440000 0x10>;
 };
@@ -51,7 +64,7 @@ sata@540000 {
 	compatible = "marvell,armada-cp110-ahci";
 	reg = <0x540000 0x30000>;
 	dma-coherent;
-	interrupts = <GIC_SPI 287 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 107 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cps_syscon0 1 15>;
 	status = "disabled";
 	port_base = <0x10000>;
@@ -62,7 +75,7 @@ usb3h0_1: usb3@500000 {
 	compatible = "generic-xhci";
 	reg = <0x500000 0x4000>;
 	dma-coherent;
-	interrupts = <GIC_SPI 286 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 106 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cps_syscon0 1 22>;
 	status = "disabled";
 };
@@ -71,7 +84,7 @@ usb3h1_1: usb3@510000 {
 	compatible = "generic-xhci";
 	reg = <0x510000 0x4000>;
 	dma-coherent;
-	interrupts = <GIC_SPI 285 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 105 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cps_syscon0 1 23>;
 	status = "disabled";
 };
@@ -101,7 +114,7 @@ i2c@701000 {
 	reg = <0x701000 0x20>;
 	#address-cells = <1>;
 	#size-cells = <0>;
-	interrupts = <GIC_SPI 310 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 120 IRQ_TYPE_LEVEL_HIGH>;
 	timeout-ms = <1000>;
 	clocks = <&cps_syscon0 1 21>;
 	status = "disabled";
@@ -112,7 +125,7 @@ i2c@701100 {
 	reg = <0x701100 0x20>;
 	#address-cells = <1>;
 	#size-cells = <0>;
-	interrupts = <GIC_SPI 311 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 121 IRQ_TYPE_LEVEL_HIGH>;
 	timeout-ms = <1000>;
 	clocks = <&cps_syscon0 1 21>;
 	status = "disabled";
@@ -126,7 +139,7 @@ nand@720000 {
 	num-cs = <1>;
 	marvell,nand-enable-arbiter;
 	nand-on-flash-bbt;
-	interrupts = <GIC_SPI 307 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 115 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cps_syscon0 1 2>;
 	nand-ecc-strength = <4>;
 	nand-ecc-step-size = <512>;
@@ -245,17 +258,17 @@ axim-hb1-wr@3c9000 {
 
 gop {
 	emac0_1: mac0 {
-		interrupts = <GIC_SPI 292 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+		interrupts = <ICU_GRP_NSR 129 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
 		mac-address = [00 00 00 00 00 04];
 		port-id = <0>; /* gop_port_id */
 	};
 	emac2_1: mac2 {
-		interrupts = <GIC_SPI 291 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+		interrupts = <ICU_GRP_NSR 128 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
 		mac-address = [00 00 00 00 00 05];
 		port-id = <2>; /* gop_port_id */
 	};
 	emac3_1: mac3 {
-		interrupts = <GIC_SPI 290 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+		interrupts = <ICU_GRP_NSR 127 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
 		mac-address = [00 00 00 00 00 06];
 		port-id = <3>; /* gop_port_id */
 	};
@@ -283,31 +296,31 @@ ppv22@000000 {
 	clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
 	status = "okay";
 	eth0_1: eth0@010000 {
-		interrupts = <GIC_SPI 261 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-			     <GIC_SPI 265 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-			     <GIC_SPI 269 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-			     <GIC_SPI 273 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-			     <GIC_SPI 316 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+		interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+			     <ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+			     <ICU_GRP_NSR 47 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+			     <ICU_GRP_NSR 51 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+			     <ICU_GRP_NSR 55 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
 		port-id = <0>; /* pp2_port_id */
 		emac-data = <&emac0_1>;
 		status = "disabled";
 	};
 	eth1_1: eth1@020000 {
-		interrupts = <GIC_SPI 262 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-			     <GIC_SPI 266 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-			     <GIC_SPI 270 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-			     <GIC_SPI 274 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-			     <GIC_SPI 317 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+		interrupts = <ICU_GRP_NSR 40 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+			     <ICU_GRP_NSR 44 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+			     <ICU_GRP_NSR 48 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+			     <ICU_GRP_NSR 52 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+			     <ICU_GRP_NSR 56 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
 		port-id = <1>; /* pp2_port_id */
 		emac-data = <&emac2_1>;
 		status = "disabled";
 	};
 	eth2_1: eth2@030000 {
-		interrupts = <GIC_SPI 263 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-			     <GIC_SPI 267 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-			     <GIC_SPI 271 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-			     <GIC_SPI 275 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-			     <GIC_SPI 318 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+		interrupts = <ICU_GRP_NSR 41 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+			     <ICU_GRP_NSR 45 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+			     <ICU_GRP_NSR 49 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+			     <ICU_GRP_NSR 53 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+			     <ICU_GRP_NSR 57 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
 		port-id = <2>; /* pp2_port_id */
 		emac-data = <&emac3_1>;
 		status = "disabled";
@@ -319,7 +332,7 @@ cps_trng: trng@760000 {
 	reg = <0x760000 0x7D>;
 	#address-cells = <1>;
 	#size-cells = <0>;
-	interrupts = <GIC_SPI 312 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 95 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cps_syscon0 1 25>;
 	status = "okay";
 };
@@ -328,7 +341,7 @@ eip197@800000 {
 	compatible = "marvell,eip197";
 	reg = <0x800000 0x200000>;
 	dma-coherent;
-	interrupts = <GIC_SPI 278 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cps_syscon0 1 26>;
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
index 75e7d80..9bee65e 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
@@ -51,7 +51,7 @@
 		#address-cells = <2>;
 		#size-cells = <2>;
 		compatible = "simple-bus";
-		interrupt-parent = <&gic>;
+		interrupt-parent = <&icu_m>;
 		ranges;
 
 		config-space@f2000000 {
@@ -373,8 +373,8 @@
 			ranges = <0x81000000 0 0xf9000000 0  0xf9000000 0 0x10000	/* downstream I/O */
 				  0x82000000 0 0xf6000000 0  0xf6000000 0 0xf00000>;	/* non-prefetchable memory */
 			interrupt-map-mask = <0 0 0 0>;
-			interrupt-map = <0 0 0 0 &gic 0 GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>;
-			interrupts = <GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-map = <0 0 0 0 &icu_m ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
 			num-lanes = <1>;
 			clocks = <&cpm_syscon0 1 13>;
 			status = "disabled";
@@ -394,8 +394,8 @@
 			ranges = <0x81000000 0 0xf9010000 0  0xf9010000 0 0x10000	/* downstream I/O */
 				  0x82000000 0 0xf7000000 0  0xf7000000 0 0xf00000>;	/* non-prefetchable memory */
 			interrupt-map-mask = <0 0 0 0>;
-			interrupt-map = <0 0 0 0 &gic 0 GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>;
-			interrupts = <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-map = <0 0 0 0 &icu_m ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
 
 			num-lanes = <1>;
 			clocks = <&cpm_syscon0 1 11>;
@@ -416,8 +416,8 @@
 			ranges = <0x81000000 0 0xf9020000 0  0xf9020000 0 0x10000	/* downstream I/O */
 				  0x82000000 0 0xf8000000 0  0xf8000000 0 0xf00000>;	/* non-prefetchable memory */
 			interrupt-map-mask = <0 0 0 0>;
-			interrupt-map = <0 0 0 0 &gic 0 GIC_SPI 33 IRQ_TYPE_LEVEL_HIGH>;
-			interrupts = <GIC_SPI 33 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-map = <0 0 0 0 &icu_m ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts = <ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
 
 			num-lanes = <1>;
 			clocks = <&cpm_syscon0 1 12>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-slave.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-slave.dtsi
index fd9a530..4cc3eee 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-slave.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-slave.dtsi
@@ -432,7 +432,7 @@
 				/* non-prefetchable memory */
 				0x82000000 0 0xfa000000 0  0xfa000000 0 0xf00000>;
 			interrupt-map-mask = <0 0 0 0>;
-			interrupt-map = <0 0 0 0 &icu_s 0 ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-map = <0 0 0 0 &icu_s ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
 			interrupts = <ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
 			num-lanes = <1>;
 			clocks = <&cps_syscon0 1 13>;
@@ -458,7 +458,7 @@
 				/* non-prefetchable memory */
 				0x82000000 0 0xfb000000 0  0xfb000000 0 0xf00000>;
 			interrupt-map-mask = <0 0 0 0>;
-			interrupt-map = <0 0 0 0 &icu_s 0 ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-map = <0 0 0 0 &icu_s ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
 			interrupts = <ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
 
 			num-lanes = <1>;
@@ -485,7 +485,7 @@
 				/* non-prefetchable memory */
 				0x82000000 0 0xfc000000 0  0xfc000000 0 0xf00000>;
 			interrupt-map-mask = <0 0 0 0>;
-			interrupt-map = <0 0 0 0 &icu_s 0 ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-map = <0 0 0 0 &icu_s ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
 			interrupts = <ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
 
 			num-lanes = <1>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 24d0487..3ce8c12 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -16,6 +16,7 @@
 /*
  * Generic Device Tree describing Marvell Armada CP-110 device
  */
+#include <dt-bindings/interrupt-controller/mvebu-icu.h>
 
 cpm_syscon0: system-controller@440000 {
 	compatible = "marvell,cp110-system-controller0",
@@ -37,6 +38,18 @@ cpm_syscon0: system-controller@440000 {
 		"cpm-usb3dev", "cpm-eip150", "cpm-eip197";
 };
 
+icu_m: interrupt-controller@1E0000 {
+	compatible = "marvell,icu";
+	reg = <0x1E0000 0x10>;
+	#interrupt-cells = <3>;
+	#size-cells = <1>;
+	#address-cells = <0>;
+	icu-spi-base = <0>;
+	interrupt-controller;
+	interrupt-parent = <&gic>;
+	gicp-spi = <0x0 0xF03F0040 0x0 0xF03F0048>;
+};
+
 rtc@284000 {
 	compatible = "marvell,armada8k-rtc";
 	reg = <0x284000 0x20>, <0x284080 0x24>;
@@ -56,10 +69,10 @@ gpio@440100 {
 	#gpio-cells = <2>;
 	interrupt-controller;
 	#interrupt-cells = <2>;
-	interrupts = <GIC_SPI 79 IRQ_TYPE_LEVEL_HIGH>,
-		   <GIC_SPI 78 IRQ_TYPE_LEVEL_HIGH>,
-		   <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>,
-		   <GIC_SPI 76 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 86 IRQ_TYPE_LEVEL_HIGH>,
+		   <ICU_GRP_NSR 85 IRQ_TYPE_LEVEL_HIGH>,
+		   <ICU_GRP_NSR 84 IRQ_TYPE_LEVEL_HIGH>,
+		   <ICU_GRP_NSR 83 IRQ_TYPE_LEVEL_HIGH>;
 };
 
 gpio@440140 {
@@ -71,17 +84,17 @@ gpio@440140 {
 	#gpio-cells = <2>;
 	interrupt-controller;
 	#interrupt-cells = <2>;
-	interrupts = <GIC_SPI 75 IRQ_TYPE_LEVEL_HIGH>,
-		   <GIC_SPI 74 IRQ_TYPE_LEVEL_HIGH>,
-		   <GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>,
-		   <GIC_SPI 72 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 82 IRQ_TYPE_LEVEL_HIGH>,
+		   <ICU_GRP_NSR 81 IRQ_TYPE_LEVEL_HIGH>,
+		   <ICU_GRP_NSR 80 IRQ_TYPE_LEVEL_HIGH>,
+		   <ICU_GRP_NSR 79 IRQ_TYPE_LEVEL_HIGH>;
 };
 
 serial@702000 {
 	compatible = "snps,dw-apb-uart";
 	reg = <0x702000 0x100>;
 	reg-shift = <2>;
-	interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 122 IRQ_TYPE_LEVEL_HIGH>;
 	reg-io-width = <1>;
 	clocks = <&cpm_syscon0 0  3>;
 	status = "disabled";
@@ -91,7 +104,7 @@ serial@702100 {
 	compatible = "snps,dw-apb-uart";
 	reg = <0x702100 0x100>;
 	reg-shift = <2>;
-	interrupts = <GIC_SPI 89 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 123 IRQ_TYPE_LEVEL_HIGH>;
 	reg-io-width = <1>;
 	clocks = <&cpm_syscon0 0  3>;
 	status = "disabled";
@@ -101,7 +114,7 @@ serial@702200 {
 	compatible = "snps,dw-apb-uart";
 	reg = <0x702200 0x100>;
 	reg-shift = <2>;
-	interrupts = <GIC_SPI 90 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 124 IRQ_TYPE_LEVEL_HIGH>;
 	reg-io-width = <1>;
 	clocks = <&cpm_syscon0 0  3>;
 	status = "disabled";
@@ -111,7 +124,7 @@ serial@702300 {
 	compatible = "snps,dw-apb-uart";
 	reg = <0x702300 0x100>;
 	reg-shift = <2>;
-	interrupts = <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 125 IRQ_TYPE_LEVEL_HIGH>;
 	reg-io-width = <1>;
 	clocks = <&cpm_syscon0 0  3>;
 	status = "disabled";
@@ -121,7 +134,7 @@ sata: sata@540000 {
 	compatible = "marvell,armada-cp110-ahci";
 	reg = <0x540000 0x30000>;
 	dma-coherent;
-	interrupts = <GIC_SPI 63 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 107 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 15>;
 	status = "disabled";
 	port_base = <0x10000>;
@@ -132,7 +145,7 @@ usb3h0: usb3@500000 {
 	compatible = "generic-xhci";
 	reg = <0x500000 0x4000>;
 	dma-coherent;
-	interrupts = <GIC_SPI 62 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 106 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 22>;
 	status = "disabled";
 };
@@ -141,7 +154,7 @@ usb3h1: usb3@510000 {
 	compatible = "generic-xhci";
 	reg = <0x510000 0x4000>;
 	dma-coherent;
-	interrupts = <GIC_SPI 61 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 105 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 23>;
 	status = "disabled";
 };
@@ -152,7 +165,7 @@ u3d@520000 {
 	 * 1: 0x121F10: Clock Source High reg: VBUS present indication in PHY SerDes lane#1
 	 */
 	reg = <0x520000 0x2000>, <0x121f10 0x40>;
-	interrupts = <GIC_SPI 60 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 102 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 24>;
 	dma-coherent;
 	status = "disabled";
@@ -161,7 +174,7 @@ u3d@520000 {
 udc@524100 {
 	compatible = "marvell,mv-udc";
 	reg = <0x524100 0x2000>;
-	interrupts = <GIC_SPI 60 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 102 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 24>;
 	dma-coherent;
 	status = "disabled";
@@ -192,7 +205,7 @@ i2c1: i2c@701000 {
 	reg = <0x701000 0x20>;
 	#address-cells = <1>;
 	#size-cells = <0>;
-	interrupts = <GIC_SPI 86 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 120 IRQ_TYPE_LEVEL_HIGH>;
 	timeout-ms = <1000>;
 	clocks = <&cpm_syscon0 1 21>;
 	status = "disabled";
@@ -203,7 +216,7 @@ i2c2: i2c@701100 {
 	reg = <0x701100 0x20>;
 	#address-cells = <1>;
 	#size-cells = <0>;
-	interrupts = <GIC_SPI 87 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 121 IRQ_TYPE_LEVEL_HIGH>;
 	timeout-ms = <1000>;
 	clocks = <&cpm_syscon0 1 21>;
 	status = "disabled";
@@ -217,7 +230,7 @@ nand@720000 {
 	num-cs = <1>;
 	marvell,nand-enable-arbiter;
 	nand-on-flash-bbt;
-	interrupts = <GIC_SPI 83 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 115 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 2>;
 	nand-ecc-strength = <4>;
 	nand-ecc-step-size = <512>;
@@ -272,7 +285,7 @@ mdio@15b000 {
 sdhci@780000 {
 	compatible = "marvell,xenon-sdhci";
 	reg = <0x780000 0x300>;
-	interrupts = <GIC_SPI 69 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 27 IRQ_TYPE_LEVEL_HIGH>;
 	clock-names = "core";
 	clocks = <&cpm_syscon0 1 4>;
 	xenon,slotno = <0>;
@@ -330,17 +343,17 @@ axim-hb1-wr@3c9000 {
 
 gop {
 	emac0: mac0 {
-		interrupts = <GIC_SPI 68 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+		interrupts = <ICU_GRP_NSR 129 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
 		mac-address = [00 00 00 00 00 01];
 		port-id = <0>; /* gop_port_id */
 	};
 	emac2: mac2 {
-		interrupts = <GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+		interrupts = <ICU_GRP_NSR 128 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
 		mac-address = [00 00 00 00 00 02];
 		port-id = <2>; /* gop_port_id */
 	};
 	emac3: mac3 {
-		interrupts = <GIC_SPI 66 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
+		interrupts = <ICU_GRP_NSR 127 IRQ_TYPE_LEVEL_HIGH>;  /* Link IRQ */
 		mac-address = [00 00 00 00 00 03];
 		port-id = <3>; /* gop_port_id */
 	};
@@ -369,31 +382,31 @@ ppv22@000000 {
 	status = "okay";
 	l4_chksum_jumbo_port = <0>;
 	eth0: eth0@010000 {
-		interrupts = <GIC_SPI 37 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-			     <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-			     <GIC_SPI 45 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-			     <GIC_SPI 49 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-			     <GIC_SPI 92 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+		interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+			     <ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+			     <ICU_GRP_NSR 47 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+			     <ICU_GRP_NSR 51 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+			     <ICU_GRP_NSR 55 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
 		port-id = <0>; /* pp2_port_id */
 		emac-data = <&emac0>;
 		status = "disabled";
 	};
 	eth1: eth1@020000 {
-		interrupts = <GIC_SPI 38 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-			     <GIC_SPI 42 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-			     <GIC_SPI 46 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-			     <GIC_SPI 50 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-			     <GIC_SPI 93 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+		interrupts = <ICU_GRP_NSR 40 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+			     <ICU_GRP_NSR 44 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+			     <ICU_GRP_NSR 48 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+			     <ICU_GRP_NSR 52 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+			     <ICU_GRP_NSR 56 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
 		port-id = <1>; /* pp2_port_id */
 		emac-data = <&emac2>;
 		status = "disabled";
 	};
 	eth2: eth2@030000 {
-		interrupts = <GIC_SPI 39 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-			     <GIC_SPI 43 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-			     <GIC_SPI 47 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-			     <GIC_SPI 51 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-			     <GIC_SPI 94 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+		interrupts = <ICU_GRP_NSR 41 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
+			     <ICU_GRP_NSR 45 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
+			     <ICU_GRP_NSR 49 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
+			     <ICU_GRP_NSR 53 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
+			     <ICU_GRP_NSR 57 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
 		port-id = <2>; /* pp2_port_id */
 		emac-data = <&emac3>;
 		status = "disabled";
@@ -405,7 +418,7 @@ cpm_trng: trng@760000 {
 	reg = <0x760000 0x7D>;
 	#address-cells = <1>;
 	#size-cells = <0>;
-	interrupts = <GIC_SPI 59 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 95 IRQ_TYPE_LEVEL_HIGH>;
 	clocks = <&cpm_syscon0 1 25>;
 	status = "okay";
 };
@@ -414,11 +427,11 @@ eip197: eip197@800000 {
 	compatible = "marvell,eip197";
 	reg = <0x800000 0x200000>;
 	dma-coherent;
-	interrupts = <GIC_SPI 54 IRQ_TYPE_LEVEL_HIGH>,
-			<GIC_SPI 55 IRQ_TYPE_LEVEL_HIGH>,
-			<GIC_SPI 56 IRQ_TYPE_LEVEL_HIGH>,
-			<GIC_SPI 57 IRQ_TYPE_LEVEL_HIGH>,
-			<GIC_SPI 58 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>,
+			<ICU_GRP_NSR 89 IRQ_TYPE_LEVEL_HIGH>,
+			<ICU_GRP_NSR 90 IRQ_TYPE_LEVEL_HIGH>,
+			<ICU_GRP_NSR 91 IRQ_TYPE_LEVEL_HIGH>,
+			<ICU_GRP_NSR 92 IRQ_TYPE_LEVEL_HIGH>;
 	interrupt-names = "ring0", "ring1", "ring2", "ring3", "global";
 	clocks = <&cpm_syscon0 1 26>;
 	status = "disabled";
-- 
1.7.9.5

